Review 2

coderbyheart opened the pull request:
- docs: update README and put most important information on top
- docs(adr): make clear that it is missing in specific object
- fix: always use permalinks
- docs: link to definition of object directly
- fix: npm package name will just be asset-tracker-lwm2m
- fix: remove warning callback
- refactor: inline functions and make docstring available
- docs: improve docstrings for error classes

coderbyheart committed:
docs: update README and put most important information on top

This also corrects some of the examples to
use the imports that they would be using.

coderbyheart committed:
docs(adr): make clear that it is missing in specific object

coderbyheart committed:
fix: always use permalinks

Linking to the saga branch will result
in the content of the page changing in the future.

coderbyheart committed:
docs: link to definition of object directly

This also removes the duplication of the object details
which is provided in the linked document

coderbyheart committed:
fix: npm package name will just be asset-tracker-lwm2m

coderbyheart committed:
fix: remove warning callback

Both callbacks only had one potential error,
and had no functional responsibility, so
they are combined in one callback which
receives two types of errors.

coderbyheart committed:
refactor: inline functions and make docstring available

This makes to docstring of the internal functions
available on the public API and reduces noise.

coderbyheart committed:
docs: improve docstrings for error classes

coderbyheart committed:
refactor: add generic converter result type

This removes code duplication.

MLopezJ commented:
# TODO

- [ ]  **Move data transition file from folder to readme**
requested: a3311b1f80b9ea8035e4181a7aaafcae32db5192
implemented:

- [ ]  **rename "documents" folder**
requested: a3311b1f80b9ea8035e4181a7aaafcae32db5192
implemented:

- [ ]  **update readme**
requested: a3311b1f80b9ea8035e4181a7aaafcae32db5192
implemented:

- [ ]  **docs(adr): make clear that it is missing in specific object**
requested: 322acf6ef740d7b8ded28519ad72c40d4cbeed85
implemented:

- [ ]  **fix: always use permalinks**
requested: 13e081dfc620174dd052689beb52e318df3f1bd3
implemented:

- [ ]  **Remove field description from docs folder**
requested: 17fd4ae2fdd6c995db90ef3ce5cad6a5c514ebaa
implemented:

- [ ]  **rename package name**
requested: bb9bf5bfa28d9271db0570f98e88d5c1d1620fe8
implemented:


- [ ]  **fix: remove warning callback**
requested: 75ecffed2608e628f51ddb17c212d297b4dbbd55
implemented:


- [ ]  **rename getTime function**
requested: 97f63f6b53cc03af31f33c610aecfc1fd7037761
implemented:

- [ ]  **remove "create..." functions**
requested: 97f63f6b53cc03af31f33c610aecfc1fd7037761
implemented:


- [ ]  ****
requested: 
implemented:

- [ ]  ****
requested: 
implemented:

MLopezJ committed:
docs: describe data transition table

MLopezJ committed:
docs: fix file link

MLopezJ committed:
fix: version used in permalinks

The file does not exist in version 31.0.0.

Version 4.1.25 was used instead.

MLopezJ committed:
fix: update type definiton

1- remove any from type declaration

2- rename type because the word 'converter' its been used in the whole document, so the intention is to follow the standard. At the end 'conversion' and 'transformation' are synonymous so I think we should pick one and follow it in the project

MLopezJ committed:
docs: move LwM2M outside

coderbyheart committed:
docs: link to nRF Connect SDK

MLopezJ commented:
WIP: implementing `nod` in `cfg`

https://github.com/MLopezJ/asset-tracker-lwm2m-js/pull/16/files#diff-77b3519370ff48f6caf7c316f0ece966f2f0871c40e9381d451d2c5b1c76b40fR17

MLopezJ committed:
test: failing test for nod

create nod arary value. 'nod' is an attribute from 'cfg' object.

MLopezJ committed:
feat: get nod value

create a list of modules which should be disabled when sampling data in nRF Asset Tracker.

nod is an array expected by Config object in nRF Asset Tracker Reported. Default value is an empty array

MLopezJ committed:
docs: fix typo
